test/e2e_env/multizone/localityawarelb/locality_aware_multizone_hybrid.go

```diff
@@ -0,0 +1,254 @@
+				}),
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-az-1-zone-4"}),
+			)).
+			Install(TestServerUniversal("test-server-node-2", mesh, WithAdditionalTags(
+				map[string]string{
+					"k8s.io/node": "node-2",
+				}),
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-node-2-zone-4"}),
+			)).
+			Install(TestServerUniversal("test-server-no-tags", mesh,
+				WithArgs([]string{"echo", "--instance", "test-server-no-tags-zone-4"}),
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+			)).
+			Setup(multizone.UniZone1),
+		).To(Succeed())
+
+		// Universal Zone 5
+		Expect(NewClusterSetup().
+			Install(DemoClientUniversal(
+				"demo-client_locality-aware-lb_svc",
+				mesh,
+				WithTransparentProxy(true),
+			)).
+			Install(TestServerUniversal("test-server", mesh,
+				WithServiceName("test-server_locality-aware-lb_svc_80"),
+				WithArgs([]string{"echo", "--instance", "test-server-zone-5"}),
+			)).
+			Setup(multizone.UniZone2),
+		).To(Succeed())
+
+		// Kubernetes Zone 1
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(democlient.WithMesh(mesh), democlient.WithNamespace(namespace))).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-zone-1"),
+			)).
+			Setup(multizone.KubeZone1)).ToNot(HaveOccurred())
+
+		// Kubernetes Zone 2
+		Expect(NewClusterSetup().
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(democlient.Install(democlient.WithMesh(mesh), democlient.WithNamespace(namespace))).
+			Install(testserver.Install(
+				testserver.WithName("test-server"),
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithEchoArgs("echo", "--instance", "test-server-zone-2"),
+			)).
+			Setup(multizone.KubeZone2)).ToNot(HaveOccurred())
+	})
+
+	E2EAfterAll(func() {
+		Expect(multizone.UniZone1.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.UniZone2.DeleteMeshApps(mesh)).To(Succeed())
+		Expect(multizone.KubeZone1.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(multizone.KubeZone2.TriggerDeleteNamespace(namespace)).To(Succeed())
+		Expect(multizone.Global.DeleteMesh(mesh)).To(Succeed())
+	})
+
+	It("should route based on defined strategy", func() {
+		Eventually(func() (map[string]int, error) {
+			return client.CollectResponsesByInstance(multizone.UniZone1, "demo-client_locality-aware-lb_svc", "test-server_locality-aware-lb_svc_80.mesh", client.WithNumberOfRequests(100))
+		}, "30s", "500ms").Should(
+			And(
+				HaveKeyWithValue(Equal(`test-server-node-1-zone-4`), BeNumerically("~", 93, 5)),
+				HaveKeyWithValue(Equal(`test-server-az-1-zone-4`), BeNumerically("~", 7, 5)),
+				Or(
```

Comment:
can you explain this or? I don't understand this assertion :(

Reply:
So most of the traffic should go to the first 2 endpoints but sometimes 1 request might reach one of 2 with the lowest weight. To avoid flakiness I've added a case that first to rule **MUST** be true, while for OR only one of them.